### PR TITLE
Disable snmpv2

### PR DESCRIPTION
**Pull request by hdep**
Hello,

Here is a small PR which create a new variable snmpv2_enable, by default this variable is at true, so there is no change from current behaviour.

If set to false, this remove com2sec, group, and access config in snmpd.conf and there is no error in the log when restarting the daemon.

### CODE REVIEW

**Comment by hdep** [comment]
Hi, 
travis is failing on snmp::client test, but I didn't touch client.pp file.
Any idea on what can explain this ? 

Regards

**Comment by hdep** [comment]
Hello @razorsedge can you help me to fix the travis error ?

File: manifests/init.pp [hunk 1]
```diff
@@ -353,6 +358,7 @@
   $agentx_socket                = $snmp::params::agentx_socket,
   $agentx_timeout               = $snmp::params::agentx_timeout,
   $agentx_retries               = $snmp::params::agentx_retries,
+  $snmpv2_enable                = $snmp::params::snmpv2_enable,
```
Comment (bastelfreak) [inline manifests/init.pp]: Can you please add a datatype to this?

File: manifests/init.pp [hunk 2]
```diff
@@ -362,6 +368,7 @@
   validate_bool($service_hasstatus)
   validate_bool($service_hasrestart)
   validate_bool($openmanage_enable)
+  validate_bool($snmpv2_enable)
```
Comment (bastelfreak) [inline manifests/init.pp]: please do not add the legacy validate_* methods. Datatypes are the new way to go \o/

**Comment by bastelfreak** [comment]
Hi @hdep, this module was recently migrated to Vox Pupuli. Please rebase against our latest master branch.  Can you please take a look at the inline comments I made and add a test for your change?

**Comment by hdep** [comment]
Do you want a new PR with the changes ? 
Since you closed this one...

**Comment by bastelfreak** [comment]
sorry, I closed it by accident! But let's continue in the new PR.

**Comment by hdep** [comment]
no problem !

**Pull request closed without merging**